server/base/src/main/java/org/apache/accumulo/server/compaction/CompactionWatcher.java

```diff
@@ -79,6 +84,10 @@ public void run() {
     copy.keySet().removeAll(newKeys);
 
     for (ObservedCompactionInfo oci : copy.values()) {
+      if (oci.stuckSample != null) {
```

Comment:
The null check / set sample to null may be unnecessary - from the micrometer docs, LongTaskTimers do not report on completed tasks - if may be enough to record the start / stop which would provide the active task count, and the active count duration info.

Reply:
A single compaction could get stuck several times over the life of the compaction. I'm trying to account for that.